x-pack/platform/plugins/shared/alerting/server/application/rule/methods/bulk_fill_gaps_by_rule_ids/batch_backfill_rule_gaps.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { clampIntervals } from '../../../../lib/rule_gaps/gap/interval_utils';
+import type { BulkFillGapsByRuleIdsParams } from './types';
+import { processAllGapsInTimeRange } from '../../../../lib/rule_gaps/process_all_gaps_in_time_range';
+import type { Gap } from '../../../../lib/rule_gaps/gap';
+import { scheduleBackfill } from '../../../backfill/methods/schedule';
+import type { BulkGapFillError } from './utils';
+import { logProcessedAsAuditEvent, toBulkGapFillError } from './utils';
+import type { RulesClientContext } from '../../../../rules_client';
+
+interface BatchBackfillRuleGapsParams {
+  rule: { id: string; name: string };
+  range: BulkFillGapsByRuleIdsParams['range'];
+  maxGapCountPerRule: number;
+}
+
+type BatchBackfillScheduleRuleGapsResult =
+  | { outcome: 'backfilled' }
+  | { outcome: 'skipped' }
+  | { outcome: 'errored'; error: BulkGapFillError };
+
+export const batchBackfillRuleGaps = async (
+  context: RulesClientContext,
+  { rule, range, maxGapCountPerRule }: BatchBackfillRuleGapsParams
+): Promise<BatchBackfillScheduleRuleGapsResult> => {
+  const logger = context.logger;
```

Comment:
nit: would it be useful to add context to this logger so it's `log.logger` is  `plugins.alerting.gaps`. or add some tags so it's easier to search for?